### ISSUE TITLE
error during build

### ISSUE THREAD

**Issue by jspaaks** [nlesc-sherlock/boatswain#5]
I'm trying to use boatswain to build https://github.com/NLeSC/xenon-docker-images/tree/a01b4547af511800f71719c5a0aff31929b424d1. here's my ``boatswain tree`` output.

```
Welcome to Boatswain
├── xenon-alpine-base
│   ├── xenon-ssh
│   └── xenon-webdav
├── xenon-phusion-base
│   ├── xenon-ftp
│   └── xenon-gridengine
├── xenon-s3
├── xenon-slurm-fixture
│   └── xenon-slurm-ssh
│       └── xenon-slurm-abstract
│           ├── xenon-slurm:14
│           ├── xenon-slurm:15
│           ├── xenon-slurm:16
│           ├── xenon-slurm:17
│           └── xenon-slurm
└── xenon-torque
```

here's a ``cat`` of the boatswain.yml file

```yml
version: 1.0
organisation: nlesc
images:
    xenon-alpine-base:
        context: xenon-alpine-base
    xenon-ssh:
        context: xenon-ssh
        from: xenon-alpine-base
    xenon-webdav:
        context: xenon-webdav
        from: xenon-alpine-base
    xenon-phusion-base:
        context: xenon-phusion-base
    xenon-ftp:
        context: xenon-ftp
        from: xenon-phusion-base
    xenon-gridengine:
        context: xenon-gridengine
        from: xenon-phusion-base
    xenon-s3:
        context: xenon-s3
    xenon-torque:
        context: xenon-torque
    xenon-slurm-fixture:
        context: xenon-slurm-fixture
    xenon-slurm-ssh:
        context: xenon-slurm-ssh
        from: xenon-slurm-fixture
    xenon-slurm-abstract:
        context: xenon-slurm-abstract
        from: xenon-slurm-ssh
    xenon-slurm:14:
        context: xenon-slurm-14
        from: xenon-slurm-abstract
    xenon-slurm:15:
        context: xenon-slurm-15
        from: xenon-slurm-abstract
    xenon-slurm:16:
        context: xenon-slurm-16
        from: xenon-slurm-abstract
    xenon-slurm:17:
        context: xenon-slurm-17
        from: xenon-slurm-abstract
    xenon-slurm:
        context: xenon-slurm-17
        from: xenon-slurm-abstract
```

I did a ``docker rmi `docker images -q` `` and have no docker containers / ``docker ps -a`` is empty.

If I then do a ``boatswain build`` I get this error and the build seems to hang:

```
Welcome to Boatswain
----------             5% (1 of 17) │████████                                                                                                                                   │ Elapsed Time: 0:00:49Traceback (most recent call last):
  File "/usr/local/bin/boatswain", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/boatswain/cli.py", line 163, in main
    force=arguments.force)
  File "/usr/local/lib/python2.7/dist-packages/boatswain/boatswain.py", line 93, in build
    verbose=verbose)
  File "/usr/local/lib/python2.7/dist-packages/boatswain/boatswain.py", line 112, in _do_action
    return self._do_action_dict(action, self.images, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/boatswain/boatswain.py", line 143, in _do_action_dict
    return self.build_list(names, images, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/boatswain/boatswain.py", line 253, in build_list
    verbose=verbose):
  File "/usr/local/lib/python2.7/dist-packages/boatswain/boatswain.py", line 359, in build_one
    ident = self._docker_progress(name, generator, verbose=verbose)
  File "/usr/local/lib/python2.7/dist-packages/boatswain/boatswain.py", line 499, in _docker_progress
    ident = json_response['aux']['Digest'][7:]
KeyError: 'Digest'
----------             5% (1 of 17) │████████       
```
(clock keeps ticking though)

Also same behavior with e.g. ``boatswain build xenon-slurm``.

could be due to how we configured boatswain or something, but it'd be nice to get some more informative feedback on what went wrong/ what to do different.

**Comment by bpmweel** [comment]
Thanks for the report!

It seems this is an issue in parsing the response from the docker engine.

I'm parsing it to get the docker hash of the built image.

Docker has the tendency to change the output of the build process even between minor versions and doesn't provide another way to get this information afaik.

Can you let me know which version of docker you're using?

I should set up a test framework to test against multiple docker versions at some point.
________________________________

**Comment by jspaaks** [comment]
```bash
docker --version
Docker version 17.06.0-ce, build 02c1d87
```

**Comment by jspaaks** [comment]
steps to reproduce error:
```bash
mkdir ~/tmp
git clone https://github.com/NLeSC/xenon-docker-images
cd xenon-docker-images
git checkout a01b4547af511800f71719c5a0aff31929b424d1
virtualenv -p /usr/bin/python2.7 .venv27
pip install --upgrade pip
pip install boatswain
boatswain tree
Welcome to Boatswain
├── xenon-alpine-base
│   ├── xenon-ssh
│   └── xenon-webdav
├── xenon-phusion-base
│   ├── xenon-ftp
│   └── xenon-gridengine
├── xenon-s3
├── xenon-slurm-fixture
│   └── xenon-slurm-ssh
│       └── xenon-slurm-abstract
│           ├── xenon-slurm:14
│           ├── xenon-slurm:15
│           ├── xenon-slurm:16
│           ├── xenon-slurm:17
│           └── xenon-slurm
└── xenon-torque
boatswain build
----------             5% (1 of 17) │████████                                                                                                                                   │ Elapsed Time: 0:00:04Traceback (most recent call last):
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/bin/boatswain", line 11, in <module>
    sys.exit(main())
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/local/lib/python2.7/site-packages/boatswain/cli.py", line 163, in main
    force=arguments.force)
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/local/lib/python2.7/site-packages/boatswain/boatswain.py", line 93, in build
    verbose=verbose)
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/local/lib/python2.7/site-packages/boatswain/boatswain.py", line 112, in _do_action
    return self._do_action_dict(action, self.images, **kwargs)
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/local/lib/python2.7/site-packages/boatswain/boatswain.py", line 143, in _do_action_dict
    return self.build_list(names, images, **kwargs)
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/local/lib/python2.7/site-packages/boatswain/boatswain.py", line 253, in build_list
    verbose=verbose):
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/local/lib/python2.7/site-packages/boatswain/boatswain.py", line 359, in build_one
    ident = self._docker_progress(name, generator, verbose=verbose)
  File "/home/daisycutter/github/nlesc/xenon-docker-images/.venv27/local/lib/python2.7/site-packages/boatswain/boatswain.py", line 499, in _docker_progress
    ident = json_response['aux']['Digest'][7:]
KeyError: 'Digest'
```
I'll see if I can dig up some more info with the debugger

**Comment by jspaaks** [comment]
I managed to track down the bug to these lines in boatswain.py:
https://github.com/nlesc-sherlock/boatswain/blob/b81d8d07a77a0e499d622820554d8f4db8251a02/boatswain/boatswain.py#L497-L499
It seems that sometimes, the ``json_response`` contains a key ``ID`` instead of a key ``Digest``. I've added code that deals with this in PR #6

**Comment by bpmweel** [comment]
I merged the pull request, commit is 060293ddffaf1a8cc61a882daba591ed35b4d10c. This issue will be fixed in the next release.
